pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/AbstractBaseAdminCommand.java

```diff
@@ -101,11 +102,14 @@ public static String sendRequest(String requestMethod, String urlString, String
     try {
       return readInputStream(conn.getInputStream());
     } catch (Exception e) {
-      return readInputStream(conn.getErrorStream());
+      if (conn.getErrorStream() != null) {
+        return readInputStream(conn.getErrorStream());
+      }
+      throw new IOException("Request failed due to connection error");
```

Comment:
Can we change this to `IOException(message, e)` to propagate the actual exception to the top?